stack-graphs/src/partial.rs

```diff
@@ -2352,18 +2354,25 @@ impl PartialPath {
                 continue;
             }
             let mut new_path = self.clone();
+            let mut new_cycle_detector = path_cycle_detector.clone();
```

Comment:
One initial intuition concern is that the new types contain `VecDeque`s, and so cloning them might be part of the reason for the increased CPU.  I wonder if there is a way to implement them with more structural sharing.  (Quite possibly not in scope for this PR)

Reply:
If this is indeed an important part of the CPU increase, that would be interesting to fix. Two options I can think of are some sort of linked list, or sticking things into an arena.

The linked list would need some `Rc` I imagine to ensure everything is cleaned up at the end. The arena might be better for cache coherence, but nothing is cleaned up until we're completely done, so higher memory usage. Would need some experimenting, I think.

A simpler optimization might be to only clone if there are multiple extensions, and otherwise keep the one value. However, I'm not sure how to write that down in Rust without duplicating the loop body for the `n = 1` case.

Reply:
Resolved now that you're using arena lists instead of `VecDeque`s